tiledb/sm/filesystem/azure.cc

```diff
@@ -173,6 +175,27 @@ Status Azure::init(const Config& config, ThreadPool* const thread_pool) {
   options.Retry.RetryDelay = retry_delay_;
   options.Retry.MaxRetryDelay = max_retry_delay;
 
+  ::Azure::Core::Http::CurlTransportOptions transport_opts;
+  SSLConfig ssl_cfg = SSLConfig(config);
+
+  if (!ssl_cfg.ca_file().empty()) {
+    transport_opts.CAInfo = ssl_cfg.ca_file();
+  }
+
+  if (!ssl_cfg.ca_path().empty()) {
+    LOG_WARN(
+        "Azure ignores the `ssl.ca_path` configuration key, "
+        "use `ssl.ca_file` instead");
+  }
+
+  if (ssl_cfg.verify() == false) {
+    transport_opts.SslVerifyPeer = false;
+  }
+
+  auto transport =
+      std::make_shared<::Azure::Core::Http::CurlTransport>(transport_opts);
```

Comment:
Mostly just because I was following the Azure docs and didn't take the time to read our make_shared implementation thoroughly enough to know if it'd ever return a different type when tracing was enabled. Re-reading it now it looks fine though so I'll update to use our version.

Reply:
Fixed.